File: tests/fixtures/sample.ts
import { Request, Response } from "express";
import axios from "axios";

function processRequest(req: Request): string {
    const query = `SELECT * FROM users WHERE id = '${req.params.id}'`;
    return query;
}

const validateInput = (input: string): boolean => {
    return input.length > 0;
};

class UserService {
    async getUser(id: number): Promise<string> {
        const result = await axios.get(`/api/users/${id}`);
        return result.data;
    }

    private sanitize(value: string): string {
        return value.trim();
    }
}

File: tests/fixtures/sample.rs
use std::io;
use std::collections::HashMap;

fn process_input(input: &str) -> String {
    let trimmed = input.trim();
    format!("processed: {}", trimmed)
}

fn validate_and_store(data: &str) {
    let result = parse_data(data);
    store_result(&result);
}

struct UserService {
    db: HashMap<String, String>,
}

impl UserService {
    fn get_user(&self, id: &str) -> Option<&String> {
        self.db.get(id)
    }
}
